products/dns/terraform.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Provider::ResourceOverrides
+  ManagedZone: !ruby/object:Provider::Terraform::ResourceOverride
+    examples: |
```

Comment:
Added literal example as the generated tests from the the example template do not work due to appending the parameter at the end of the domain name. Reverted to original (non-generated) tests.

Reply:
What do you mean by "appending the parameter at the end of the domain name"?

Reply:
I'd like to generate a test like the original:
```
resource "google_dns_managed_zone" "foobar" {
        name = "mzone-test-%s"
        dns_name = "tf-acctest-%s.hashicorptest.com."
        description = "%s"
        labels = {
                foo = "bar"
        }
```
However, due to the way variables are handled in the autogenerated tests, I could only generate a test like:
```
resource "google_dns_managed_zone" "foobar" {
        name = "mzone-test-%s"
        dns_name = "tf-acctest.hashicorptest.com.%s"
        description = "%s"
        labels = {
                foo = "bar"
        }
```
Note the placement of the ` %s` in the dns_name parameter. Any way I have have the generated tests pass a random parameter at the right place?

Reply:
Oh, I see what you're saying. I don't think so. @rileykarson?

Reply:
It's complicated. We have a limited set of nameservers that we use to host the managed public zones. For the first few customer which try to create a domain, we just give them a nameserver set and return them as `nameServers`. Once we get close to the limit, we require customers to verify the domain (and we save the last set to be used only by going through support). We actually can have this collisions at the second level domain, so for the sake of comparison, customers creating `tf-acctest-1.hashicorptest.com.` and `tf-acctest-2.hashicorptest.com.` would compete for the same set of namespaces. So ideally, the tests should be able to create a domain like `hashicorptest-$RANDOM.com.` to be fully safe.

That is also why I'd like to not have hashicorptest.com in the public docs (having different domains for acceptance tests and docs), so that people copy & pasting the docs wouldn't take the nameservers for the hashicorptest.com domain and make them unavailable to whoever is actually running acceptance tests.

Reply:
Makes sense!

How about we add some variant of `prod.mydomain.com.` as the canonical example, and [add an id from the random provider like in SslCertificate](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/examples/ssl_certificate_random_provider.tf.erb#L5). For the test only, we can then [substitute the HashiCorp domain like we do for some paths already](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform/custom_code.rb#L149) (feel free to add a TODO for me to rename the method to something more general, or do it yourself)

So we'd write something like

```hcl
dns_name = "prod-${random_integer.domain_suffix.result}.mydomain.com."
```

and turn it into this for tests:

```hcl
dns_name = "tf-acctest-${random_integer.domain_suffix.result}.hashicorptest.com."
```

Reply:
I reintroduced the examples with the random_id reference and now I'm happy with the autogenerated basic test. It passes acceptance:
```
$ make testacc TESTARGS='-run=TestAccDnsManagedZone_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccDnsManagedZone_ -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-google/version.ProviderVersion=acc"
?   	github.com/terraform-providers/terraform-provider-google	[no test files]
=== RUN   TestAccDnsManagedZone_dnsManagedZoneBasicExample
=== PAUSE TestAccDnsManagedZone_dnsManagedZoneBasicExample
=== RUN   TestAccDnsManagedZone_update
=== PAUSE TestAccDnsManagedZone_update
=== RUN   TestAccDnsManagedZone_importWithProject
=== PAUSE TestAccDnsManagedZone_importWithProject
=== CONT  TestAccDnsManagedZone_dnsManagedZoneBasicExample
=== CONT  TestAccDnsManagedZone_update
=== CONT  TestAccDnsManagedZone_importWithProject
--- PASS: TestAccDnsManagedZone_importWithProject (8.64s)
--- PASS: TestAccDnsManagedZone_dnsManagedZoneBasicExample (9.18s)
--- PASS: TestAccDnsManagedZone_update (13.05s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	(cached)
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-google/scripts	(cached) [no tests to run]
?   	github.com/terraform-providers/terraform-provider-google/version	[no test files]
```